### PR TITLE
user challenge validations completed [MP-32]

### PR DESCRIPTION
**Pull request by BernardSimpson**
feat: UserChallenge updated.
feat: completed, to show date of completion.
feat: user_notes with maximum of 1500 characters,
feat: res_notes with maximum of 15000 characters,
feat: rating challenge with integer between 1-5

### CODE REVIEW

File: app/models/user_challenge.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 class UserChallenge < ApplicationRecord
-end
+    validates :completed, presence: true
+    validates :user_notes, presence: true, length: { maximum: 1500, }
```
Comment (CraigDoesCode) [inline app/models/user_challenge.rb:3]: if you check this schema, the user_notes has a different name

File: app/models/user_challenge.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
...
+    validates :res_feedback, presence: true, length: { maximum: 1500, }
```
Comment (CraigDoesCode) [inline app/models/user_challenge.rb:4]: if you check the schema res_feedback has a different name

**Comment by CraigDoesCode** [comment]
At the moment of creation of a user challenge do note feedback and rating need to be present?

**Comment by BernardSimpson** [comment]
requested a new pull request

**Pull request closed without merging**